lib/utils/import/simplenote/index.ts

```diff
@@ -42,7 +42,14 @@ class SimplenoteImporter extends EventEmitter {
         return;
       }
 
-      const dataObj = JSON.parse(fileContent);
+      let dataObj;
+      try {
+        dataObj = JSON.parse(fileContent);
+      } catch (error) {
+        this.emit('status', 'error', 'Invalid json file.');
+        return;
+      }
```

Comment:
while we're at it, since we still leave open the possibility of runtime crashes after this `try/catch` maybe we should consider adding a real parser to the input documents so that we only allow valid documents or valid lists of notes.

```ts
const parseJson = (input: string, default: unknown) => {
	try {
		return JSON.parse(input);
	} catch (e) {
		return default;
	}
}

const parseNote = (data: unknown, defaults: Partial<T.Note>) => {
	if (null === data || 'object' !== typeof data) {h
		return null;
	}

	…

	return {
		...defaults,
		…,
		modificationDate: convertDate(data.modificationDate || data.lastModified),
		…
	}
}

const parseExportData = (input: string) => {
	const data = parseJson(input, null);
	if (null === data || 'object' !== typeof data) {
		return null;
	}

	const {activeNotes, trashedNotes} = data;
	if ((activeNotes && !Array.isArray(activeNotes)) || (trashedNotes && !Array.isArray(trashedNotes))) {
		return null;
	}

	return [
		…(activeNotes || []).map(note => parseNote(note, {isTrashed: false})).filter(a => null !== a),
		…(trashedNotes || []).map(note => parseNote(note, {isTrashed: true})).filter(a => null !== a)
	]
}
```

can't remember if `isTrashed` is the right property of the note. point is, we can do a full parse and prevent errors from creeping deeper into the app if we eliminate them here. we had this problem before with a very old account whose note object didn't contain the expected properties and the app failed to load - whitescreened it.

Reply:
I would like to address this in another PR as we want to get this fix in before Friday and I am constrained on time this week.